### ISSUE TITLE
Various Quiz issues

### ISSUE THREAD

**Issue by ReckoningReckoner** [qlicker/qlicker#354]
- UI should update for students on the courses tab which indicates that a quiz, not a session is being run.
- Students should be able to edit a submission before pressing the "done" button during a quiz? A more consistent approach would be for students to save their quiz responses automatically (i.e. incase their browser crashes) but they can only submit their answers all at once by the end.

**Comment by ryanmartinneutrino** [comment]
Yes - we should in general update SessionListItem to show more relevant information, e.g. the grade item if the session is done, remove the question progress bar which is confusing, and show if the session is of type quiz.
For submitting a quiz, I agree, it's better to save the questions and then submit them all at once. The current implementation doesn't do this because it requires a substantial change to QuestionDisplay which then needs to be able to save a response but flag it as unsaved. 
The red in the professor course is to indicate that the quiz is not reviewable by students; clicking allow review removes the red. Maybe that's confusing... I thought it important to flag sessions that students cannot review, but we can probably do this more gracefully when updating sessionListItem.

**Comment by ReckoningReckoner** [comment]
In terms of the the flagged review session, I did find the red sessionListItem to be inconsistent. For instance, the "red" while grading indicates that a grade cannot be finished, but it should not also mean that it is not in review.

In my opinion, the "allow review" button already gives an indication as to whether the session is reviewable or not. If more indication is needed, then clicking the "allow review" should change the state from "FINISHED" to something like "REVIEWABLE", and should accordingly be selectable in the session toolbar. A sample flow might be like "Professor creates and runs session", "Session appears red because it is not fully graded", "A professor sets the session to REVIEWABLE after all the grades are fully marked."
